src/Frame/Frame.cc

```diff
@@ -480,6 +483,11 @@ bool Frame::FillRasterTileData(CARTA::RasterTileData& raster_tile_data, const Ti
         raster_tile_data.clear_tiles();
     }
 
+    if (!_tile_pool) {
+        _tile_pool = std::make_shared<TilePool>();
+        _tile_pool->Grow(num_threads);
+    }
```

Comment:
This isn't threadsafe, surely -- multiple threads may try to create the pool at once.

`Grow` doesn't allocate any memory -- it just sets the capacity, which is the upper bound of tiles to keep. `Pull` creates tiles on demand if there are no recycled tiles available in the pool. So it should be fine to create the pool and set its capacity in the constructor.